packages/frontend/component/src/ui/modal/confirm-modal.tsx

```diff
@@ -36,13 +42,69 @@ export const ConfirmModal = ({
       <div
         className={clsx(styles.modalFooter, {
           modalFooterWithChildren: !!children,
+          reverse: reverseFooter,
         })}
       >
         <DialogTrigger asChild>
-          <Button {...cancelButtonOptions}>{cancelText}</Button>
+          <Button onClick={onCancel} {...cancelButtonOptions}>
+            {cancelText}
+          </Button>
         </DialogTrigger>
         <Button onClick={onConfirm} {...confirmButtonOptions}></Button>
       </div>
     </Modal>
   );
 };
+
+interface ConfirmModalContextProps {
+  modalProps: ConfirmModalProps;
+  setModalProps: Dispatch<SetStateAction<ConfirmModalProps>>;
+  openConfirmModal: (props?: ConfirmModalProps) => void;
+  closeConfirmModal: () => void;
+}
+const ConfirmModalContext = createContext<ConfirmModalContextProps>({
+  modalProps: { open: false },
+  setModalProps: () => {},
+  openConfirmModal: () => {},
+  closeConfirmModal: () => {},
+});
+export const ConfirmModalProvider = ({ children }: PropsWithChildren) => {
+  const [modalProps, setModalProps] = useState<ConfirmModalProps>({
+    open: false,
+  });
+
+  const openConfirmModal = useCallback((props?: ConfirmModalProps) => {
+    setModalProps({ ...props, open: true });
+  }, []);
+
+  const closeConfirmModal = useCallback(() => {
+    setModalProps({ open: false });
+  }, []);
+
+  const onOpenChange = useCallback((open: boolean) => {
+    setModalProps(props => ({ ...props, open }));
+  }, []);
+
+  return (
+    <ConfirmModalContext.Provider
+      value={{ openConfirmModal, closeConfirmModal, setModalProps, modalProps }}
+    >
+      {children}
+      {/* TODO: multi-instance support(unnecessary for now) */}
+      <ConfirmModal onOpenChange={onOpenChange} {...modalProps} />
+    </ConfirmModalContext.Provider>
+  );
+};
+export const useConfirmModal = () => {
+  const context = useContext(ConfirmModalContext);
+  if (!context) {
+    throw new Error(
+      'useConfirmModal must be used within a ConfirmModalProvider'
+    );
+  }
+  return {
+    openConfirmModal: context.openConfirmModal,
```

Comment:
Can we merge `setModalProps` into `openConfirmModal`?

Reply:
Actually almost the same, but I think separating them can avoid unnecessary ambiguity.